src/Microsoft.ML.Data/Utilities/StreamUtils.cs

```diff
@@ -174,5 +174,39 @@ private static string[] Expand(string pattern)
             return matchList.ToArray();
         }
 #endif
+
+        public static string ReadEntry(this TextReader sr)
+        {
+            string entry = string.Empty;
+
+            // get first bit
+            entry += sr.ReadLine();
+
+            // And get more lines until the number of quotes is even
+            while (GetNumberOf(entry, "\"") % 2 != 0 )
+            {
+                string line = sr.ReadLine();
+                entry += line;
```

Comment:
Likely want to begin a string builder instead of a `+=`.

Reply:
You'll want to push thru a newline into the string builder. `ReadLine()` doesn't include the `\n` or `\r` or `\r\n` [[ref](https://docs.microsoft.com/en-us/dotnet/api/system.io.streamreader.readline?view=netcore-3.0#remarks)].

Otherwise:
```csv
a,b,"c my missing
space",e,f
``` 

Would be read as:
```csv
a,b,"c my missingspace",e,f
```

Instead of:
```csv
a,b,"c my missing\nspace",e,f
```

It likely is not important, but it may be better to match the `\n` or `\r` or `\r\n` of the last line read. This would only be important if the user read the file w/ `TextLoader` but fed the model with an IEnumerable.